nautobot_ssot/integrations/librenms/diffsync/adapters/librenms.py

```diff
@@ -67,41 +75,187 @@ def load_location(self, location: dict):
+                        return
+
                 else:
-                    _status = librenms_status_map[device["status"]]
-                new_device = self.device(
-                    name=normalize_device_hostname(device[self.hostname_field]),
-                    device_id=device["device_id"],
-                    location=(device["location"] if device["location"] is not None else "Unknown"),
-                    role=device["type"] if device["type"] is not None else None,
-                    serial_no=device["serial"] if device["serial"] is not None else "",
-                    status=_status,
-                    manufacturer=(
-                        os_manufacturer_map.get(device["os"])
-                        if os_manufacturer_map.get(device["os"]) is not None
-                        else "Unknown"
-                    ),
-                    device_type=(device["hardware"] if device["hardware"] is not None else "Unknown"),
-                    platform=device["os"] if device["os"] is not None else "Unknown",
-                    os_version=(device["version"] if device["version"] is not None else "Unknown"),
-                    ip_address=device["ip"],
-                    system_of_record=os.getenv("NAUTOBOT_SSOT_LIBRENMS_SYSTEM_OF_RECORD", "LibreNMS"),
-                )
+                    device_validation_dict = {
+                        self.job.hostname_field: normalized_name,
+                        "location": {"valid": False, "reason": "Hostname validation failed"},
+                        "role": {"valid": False, "reason": "Hostname validation failed"},
+                        "platform": {"valid": False, "reason": "Hostname validation failed"},
+                        "device_type": {"valid": False, "reason": "Hostname validation failed"},
+                    }
+                    validation_result = has_required_values(device_validation_dict, self.job)
+                    device["field_validation"] = validation_result
+
+                    if self.job.debug:
+                        self.job.logger.debug(f"Validation result for {device[hostname_field]}: {validation_result}")
+
+                    # Hostname validation failed, so this device should always fail
+                    failed_device = device.copy()
+                    self.failed_import_devices.append(failed_device)
+                    return
+
                 try:
-                    self.add(new_device)
-                except ObjectAlreadyExists:
-                    self.job.logger.warning(f"Device {device[self.hostname_field]} already exists. Skipping.")
+                    self.get(self.device, {"name": normalized_name})
+                except ObjectNotFound:
+                    if device["disabled"] == 1:
+                        _status = "Offline"
+                    else:
+                        # Handle case where device["status"] might be a Status object
```

Comment:
This doesn't make sense. Are you saying that the API response somehow has a Status object from Nautobot??

Reply:
No, it doesn't.  It only uses the incoming disabled/status values for a mapping.  Removed the code handling a "NB Status" being passed in.  That was just wrong and I should have caught it prior to this.